### DISCUSSION TITLE
Docker Metric

### DISCUSSION THREAD

**Discussion by lanrekkeeg** [fluent/fluent-bit#5023] [Q&A]
Is there a way to convert the docker metric ```cpu_used``` value to percentage in docker metric plugin? because this value doesn't make any sense until converted to percentage. Any formula or workaround will be appreciated.

[1] docker.0: [1571994772.00555745, {"id"=>"6bab19c3a0f9", "name"=>"postgresql", **"cpu_used"=>172102435**, "mem_used"=>5693400, "mem_limit"=>4294963200}]

**Comment by patrick-stephens** [comment]
Lua filter? Maybe a script that updates it that way?

**Reply by lanrekkeeg** [reply]
A script is like rewriting the plugin because then i need to read cgroup value in order to calculate the percentage. I am not sured why it is not done in first place like what is the usecase of getting this cpu value in long number?

**Reply by agup006** [reply]
Probably worth creating a new plugin based on Prometheus standards here, @lanrekkeeg is that something you would be open to contribute?

**Reply by lanrekkeeg** [reply]
yes, we need to create this new plugin accordingly. I am willing to contribute from next month (July) onward.

**Reply by agup006** [reply]
@lanrekkeeg awesome if you can contribute, I would recommend looking at the NGINX metric plugin as a starter.

Additionally, you may be able to get away with using the Prometheus Scraper and configuring Docker to expose Prometheus Metrics: https://docs.docker.com/config/daemon/prometheus/

**Reply by agup006** [reply]
@lanrekkeeg would this still be something you would be open to contribute?

**Reply by lanrekkeeg** [reply]
@agup006 Yes, I had 2 exams left which I will be giving this month. After this, I will start working on this. I will look into your suggestion.

**Reply by ypo777** [reply]
**Hi maintainers — any update on this?**  
I’m running into the same issue where Docker metrics’ `cpu_used` is a raw counter and not directly usable as a percentage.

**Context:** Fluent Bit **v4.1.1**, input name **docker (metrics)** → `cpu_used`, `mem_used`, `mem_limit`.

**Example log:**
```text
Oct 14 12:34:19 homelab fluent-bit[3187491]: [6] docker.metrics: [[1760445258.995845997, {}], {"id"=>"04b00eb1d665", "name"=>"docker-registry", "log"=>"Oct 14 12:34:18 - docker-registry -nil", "cpu_used"=>3023971, "mem_used"=>14323712, "mem_limit"=>-9223372036854775808, "environment"=>"dev"}]
```

Is there any progress on exposing a `cpu_percent` metric (Prometheus-style) or guidance on the recommended conversion? I can work around it with a Lua filter (delta over interval and CPU count), but I’d prefer an official solution.

Thank you!
